### PR TITLE
[Synthetics] Add support to dnsServer and client certificate options

### PR DESCRIPTION
**Pull request by romainberger**
This PR updates the go-client to add support to `dnsServer` and client certificate for synthetics tests.

### CODE REVIEW

File: datadog/resource_datadog_synthetics_test_.go [hunk 1]
```diff
@@ -231,6 +242,34 @@ func syntheticsTestRequest() *schema.Schema {
 					Optional: true,
 					Default:  60,
 				},
+				"dnsServer": {
```
Comment (therve) [inline datadog/resource_datadog_synthetics_test_.go:245]: In this case we can easily hide the API case, let's rename it to `dns_server`.

File: datadog/resource_datadog_synthetics_test_.go [hunk 2]
```diff
@@ -231,6 +242,34 @@ func syntheticsTestRequest() *schema.Schema {
...
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"filename": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"updated_at": {
+					Type:     schema.TypeString,
+					Required: true,
```
Comment (therve) [inline datadog/resource_datadog_synthetics_test_.go:271]: Are those last 2 required? Not sure why the API case about filename, or what updated_at means.

**Review by therve — CHANGES_REQUESTED** [review]
Looks good, some minor comments

**Review by PierreGuyot — DISMISSED** [review]

File: datadog/resource_datadog_synthetics_test_.go (hunk 2 above)
Comment (romainberger) [reply datadog/resource_datadog_synthetics_test_.go:271]: Yes that's two fields that make sense when using the UI but not really with terraform, I'll make the `updatedAt` option automatically fill in the backend, and see what we can do for `filename`.

Comment (romainberger) [reply datadog/resource_datadog_synthetics_test_.go]: I've removed the `updatedAt` option and we decided to automatically fill the `filename` (required in the backend) so it makes more sense.

**Review by therve — APPROVED** [review]

**Pull request merged**